### DISCUSSION TITLE
Question for felllow Oqtane Developers - SignalR

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#3159] [Q&A]
Has anyone been able to implement it in a module?

My code compiles, no problem but when I run it 💣 !

Error:
`An Unexpected Error Has Occurred In StudioElf.VisitorCloud, StudioElf.VisitorCloud.Client.Oqtane: Could not load file or assembly 'Microsoft.AspNetCore.SignalR.Client.Core, Version=7.0.10.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.`

Removed and re Added the package and also when down a version to 7.0.5, still the same error.

**Comment by wcrisafu** [comment]
Not sure if this is the same problem but a few months ago I needed a quick way to broadcast to messages to all connected clients. In order to use SignalR I had to modify Oqtane itself (which I do not like to do). 

This is the patch used:
Oqtane.Server -> Infrastructure -> TenantManager.cs (look for the "WMC" comment)
>>>
                var httpcontext = _httpContextAccessor.HttpContext;
                if (httpcontext != null)
                {
                    // legacy support for client api requests which would include the alias as a path prefix ( ie. {alias}/api/[controller] )
                    int aliasId;
                    string[] segments = httpcontext.Request.Path.Value.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
                    if (segments.Length > 1 && Shared.Constants.ReservedRoutes.Contains(segments[1]) && int.TryParse(segments[0], out aliasId))
                    {
                        alias = _aliasRepository.GetAliases().ToList().FirstOrDefault(item => item.AliasId == aliasId);
                    }
                    //WMC: Keep till I find another solution
                    else if (segments.Length > 1 && segments[0] == "tmshub")
                    {
                        aliasId = 1;
                    }
                    //
<<<
In my ServerStartup -> under "ConfigureServices"
>>>
services.AddSignalR();
services.AddResponseCompression(opts =>
            {
                opts.MimeTypes = ResponseCompressionDefaults.MimeTypes.Concat(
                    new[] { "application/octet-stream" });
            });
<<<
Under "Configure"
>>>
           app.UseResponseCompression();
            app.UseRouting();
            app.UseEndpoints(endpoints =>
            {
                endpoints.MapHub<TMSHub>("/tmshub");

            });
<<<

**Comment by Sen-Gupta** [comment]
Was it possible to do it?
